src/SignalR/clients/ts/signalr/src/HubConnection.ts

```diff
@@ -672,26 +674,57 @@ export class HubConnection {
         this.connection.stop(new Error("Server timeout elapsed without receiving a message from the server."));
     }
 
-    private _invokeClientMethod(invocationMessage: InvocationMessage) {
+    private async _invokeClientMethod(invocationMessage: InvocationMessage) {
         const methods = this._methods[invocationMessage.target.toLowerCase()];
-        if (methods) {
-            const methodsCopy = methods.slice();
-            try {
-                methodsCopy.forEach((m) => m.apply(this, invocationMessage.arguments));
-            } catch (e) {
-                this._logger.log(LogLevel.Error, `A callback for the method ${invocationMessage.target.toLowerCase()} threw error '${e}'.`);
-            }
+        if (!methods) {
+            this._logger.log(LogLevel.Warning, `No client method with the name '${invocationMessage.target.toLowerCase()}' found.`);
 
+            // No handlers provided by client but the server is expecting a response still, so we send an error
             if (invocationMessage.invocationId) {
-                // This is not supported in v1. So we return an error to avoid blocking the server waiting for the response.
-                const message = "Server requested a response, which is not supported in this version of the client.";
-                this._logger.log(LogLevel.Error, message);
+                this._logger.log(LogLevel.Warning, `No result given for '${invocationMessage.target.toLowerCase()}' method and invocation ID '${invocationMessage.invocationId}'.`);
+                await this._sendWithProtocol(this._createCompletionMessage(invocationMessage.invocationId, "Client didn't provide a result.", null));
+            }
+            return;
+        }
 
-                // We don't want to wait on the stop itself.
-                this._stopPromise = this._stopInternal(new Error(message));
+        // Avoid issues with handlers removing themselves thus modifying the list while iterating through it
+        const methodsCopy = methods.slice();
+
+        // Server expects a response
+        if (invocationMessage.invocationId) {
+            // We preserve the last result or exception but still call all handlers
+            let res;
+            let exception;
+            for (const m of methodsCopy) {
+                try {
+                    if (res || exception) {
+                        this._logger.log(LogLevel.Warning, `Result already provided for '${invocationMessage.target.toLowerCase()}' only the last one will be sent.`);
```

Comment:
Can we have this result in an exception on the server instead of using the last value?